pkg/trait/mount.go

```diff
@@ -118,13 +119,15 @@ func (t *mountTrait) Apply(e *Environment) error {
 		return err
 	}
 
-	// CronJob
-	if err := e.Resources.VisitCronJobE(func(cron *v1beta1.CronJob) error {
-		volumes = &cron.Spec.JobTemplate.Spec.Template.Spec.Volumes
-		visited = true
-		return nil
-	}); err != nil {
-		return err
+	if ok, err := kubernetes.IsAPIResourceInstalled(e.Client, batchv1.SchemeGroupVersion.String(), reflect.TypeOf(batchv1.CronJob{}).Name()); ok && err == nil {
```

Comment:
Is this really necessary? My understanding is that there is no server API requests made while visiting the resources?

Reply:
yes let's try without. makes life in tests much easier